script.js

```diff
@@ -2,14 +2,39 @@ const container = document.getElementById('grid');
 const resetButton = document.getElementById('reset');
 
 function buildGrid(size) {
+    /**
+     * These nested loops will work fine for small numbers, because computers are good
+     * at executing these simple calculations quickly, but generally speaking, try to
+     * avoid more loops than are needed
+     *
+     * Talk it out in English as to what's happening here:
+     * 1. We have a size param, that defaults to 16 or whatever
+     * 2. Start our first for loop, and increment until we're less than / equal to our size (16)
+     * 3. Ok I'm on my first iteration (1) of my outer i loop - now what?
+     *      a.) We'll start another for loop, and increment until we're less than / equal to our size (16)
+     *      b.) Ok I'm on my first iteration (1) of my inner j loop - now what?
+     *          i.) Well, let's create a div and give it some CSS, and then append it to our grid!
+     *      c.) Cool now what? Well I've finished everything in this first j iteration (1), so let's go back up
+     *      d.) Ok now I'm on my second iteration (2) of my inner j loop - now what?
+     *          i.) I'm going to repeat the same steps of creating a div, giving it some CSS, and appending it
+     *              to my grid container
+     *      e.) Cool now what? I'm done, so let's go on to the third iteration (3) of my inner j loop
+     *      f.) Repeat 16 times until we meet our size condition (j <= size). Then, once that is done, I'm done
+     *          with my inner loop - now what? Well, we gotta go back up to the inner loop, Step 3
+     * 4. Ok I've completed the first iteration (1) of my outer i loop - now what?
+     * 5. Well, let's start on the second iteration (2) of my outer i loop, which will...
+     *    a.) Repeat the inner j loop all over again!
+     *
+     * Can you think of a way we might achieve this in a single loop?
+     */
```

Comment:
The only thing that comes to mind would be to build a row of cells first, and then use the loop to append it for columns? I'm still working on this logic.

Reply:
That's certainly one approach! But also, take a look at the output that you're rendering to the DOM:

![Screen Shot 2021-08-14 at 9 20 46 AM](https://user-images.githubusercontent.com/10368638/129447645-ef2d954a-7141-44a5-80dd-67ae2c63f62d.png)

We don't have the concept of "row" `div`s here, because CSS grid is taking care of that for us. _Before_ grid (and flexbox for that matter too), it would be very difficult to achieve this layout without row `div`s.

Reply:
I think I'm seeing the resolution. Originally, for whatever reason, I'm thinking I have to build a specific layout with the loop. But thats what CSS is going to do for me!

`for(let i =0; i <=size*size; i++) {

//build that grid...
`